skopt/searchcv.py

```diff
@@ -463,14 +510,33 @@ def fit(self, X, y=None, *, groups=None, callback=None, **fit_params):
         else:
             self.optimizer_kwargs_ = dict(self.optimizer_kwargs)
 
+        if callable(self.refit):
+            raise ValueError("BayesSearchCV doesn't support callable refit")
+
+        # Adapted from BaseSearchCV fit() method
+        if callable(self.scoring):
+            # will be determined later
+            self._target_score = None
+        elif self.scoring is None or isinstance(self.scoring, str):
+            self._target_score = "score"
+        else:
+            # proper checking will be performed in BaseSearchCV.fit()
+            self._target_score = self.refit
```

Comment:
I ended up moving this to `_run_search`, with a bit of caching to avoid constantly computing all score names and checking `self.refit` in the case of a callable `refit`